### PR TITLE
Luminescent eyes no longer toggle back on with movement. Colour switching keeps light on.

### LINKED ISSUE #54841: High Luminoscity Eyes cannot be turned off
## Reproduction:
Implant eyes, turn them on, cannot turn them off

* If you turn them off and you don't move they will remain off, but when you move they're magically back on without a notification.

### PR DESCRIPTION
**Pull request by prodirus**
## About The Pull Request

Luminescent eyes no longer immediately toggle back on during movement, if turned on/off previously.

Light colour/distance now immediately updates when changed, without having to retoggle the lights.

output

## Why It's Good For The Game

Fixes #54841
Also, QoL change for changing colours/distance.

## Changelog
:cl:
tweak: Luminescent eyes can now alter their color and distance while on without turning themselves off.
fix: Luminescent eyes should now stop turning on again on movement!
/:cl:

### CODE REVIEW

**Comment by prodirus** [comment]
I think this could be done better, but I'm dumb newbab shitcoder. Open for criticism/suggestions!

File: code/modules/surgery/organs/eyes.dm [hunk 1]
```diff
@@ -251,15 +251,17 @@
...
 	var/range = input(user, "Enter range (0 - [max_light_beam_distance])", "Range Select", 0) as null|num
-
-	set_distance(clamp(range, 0, max_light_beam_distance))
 	assume_rgb(C)
+	// Reactivate if eyes were already active for real time colour swapping!
+	if(active)
+		set_distance(clamp(range, 0, max_light_beam_distance))
+		activate(FALSE)
+	else
+		set_distance(clamp(range, 0, max_light_beam_distance))
```
Comment (Jared-Fogle) [inline code/modules/surgery/organs/eyes.dm:260]: This is a duplicate line, move it out and then just do `if(active) activate(FALSE)`

Comment (prodirus) [reply code/modules/surgery/organs/eyes.dm:260]: set_distance() would set active to false, but I can move the false assignment and it should work. Will do some testing before I nuke this dupe line.

Comment (Jared-Fogle) [reply code/modules/surgery/organs/eyes.dm]: You can also just store the value beforehand, `var/old_active = active` for example.

Comment (prodirus) [reply code/modules/surgery/organs/eyes.dm]: There, thanks for the advice!

**Pull request merged**